src/sdam/topology.ts

```diff
@@ -459,12 +458,19 @@ export class Topology extends TypedEventEmitter<TopologyEvents> {
     }
 
     const timeoutMS = this.client.options.timeoutMS;
-    const timeout = timeoutMS != null ? Timeout.expires(timeoutMS) : undefined;
+    const serverSelectionTimeoutMS = this.client.options.serverSelectionTimeoutMS;
     const readPreference = options.readPreference ?? ReadPreference.primary;
+
+    const timeoutContext = TimeoutContext.create({
+      timeoutMS,
+      serverSelectionTimeoutMS,
+      waitQueueTimeoutMS: this.client.options.waitQueueTimeoutMS
+    });
+
     const selectServerOptions = {
       operationName: 'ping',
-      timeout,
-      ...options
+      ...options,
+      timeoutContext
```

Comment:
Note: We construct a `TimeoutContext` here as we treat the ping command used for autoconnect as its own command.

Reply:
I think we can consider this more fully later, but wouldn't that mean that in the case that an operation auto-connects it's possible for the operation to take up to 2x timeoutMS?

Reply:
I filed [NODE-6223](https://jira.mongodb.org/browse/NODE-6223) to handle auto connect, we can worry about what the correct behavior is later if that works for you?